File: frontend/src/types/media.ts
import { MediaType } from ".";

/** A media item from the library */
export interface MediaItem {
    id: number;
    external_id: string;
    tmdb_id: number | null;
    title: string;
    type: MediaType;
    year: number | null;
    genres: string[] | null;
    quality: string | null;
    size_bytes: number | null;
    poster_url: string | null;
    source_service: string;
    added_at: string | null;
    synced_at: string;
}

File: backend/app/schemas/media.py
"""Media schemas."""

from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class MediaItemResponse(BaseModel):
    """Media item response."""
    id: int
    external_id: str
    tmdb_id: int | None = None
    title: str
    type: str
    year: int | None = None
    genres: list[str] | None = None
    quality: str | None = None
    size_bytes: int | None = None
    poster_url: str | None = None
    source_service: str
    added_at: datetime | None = None
    synced_at: datetime

    model_config = {"from_attributes": True}
